frontend/app/questions/addQuestionModal.tsx

```
@@ -38,14 +39,27 @@ export default function QuestionAddModal() {
   } = useForm();
 
   const onSubmit = handleSubmit(async (data: Question) => {
-    const modifiedData = {
-      ...data,
-      categories: (data.categories as string).split(',') as Category[],
-    };
-    dispatch(addQuestion(modifiedData));
-    notifyAdd();
-    onOpenChange();
-    reset();
+      const modifiedData = {
+        ...data,
+        categories: (data.categories as string).split(',') as Category[],
+      };
+      try {
+        await axiosInstance.post('', modifiedData);
+        setUpdate(true);
+        notifyAdd();
+      } catch(error) {
+        const status = error.response.status;
+        if (status === 400) {
+          notifyError("Bad Request: Ensure data inputted is valid");
+        } else if (status === 409) {
+          notifyError("Conflict: This question already exists");
+        } else {
+          notifyError("An error occurred. Please try again later");

```

backend should return the correct error message, so on the frontend, you don't have to check every single status code, just return the error message created by the backend if there is an error